src/gui/preferences.c

```diff
@@ -1104,9 +1122,243 @@ static void edit_preset_response(dt_gui_presets_edit_dialog_t *g)
+  {
+    g_free(ith_name);
+    g_free(ith_operation);
+
+    GtkTreeIter parent;
+    gtk_tree_model_iter_parent(tree_model, &parent, iter);
+
+    gtk_tree_store_remove(GTK_TREE_STORE(tree_model), iter);
+
+    // check whether the parent has children; if not, remove the parent
+    gint n_children = gtk_tree_model_iter_n_children (tree_model, &parent);
+    if(! n_children) gtk_tree_store_remove(GTK_TREE_STORE(tree_model), &parent);
+
+    return TRUE;
+  }
+  return FALSE;
+}
+
+static gboolean delete_by_name_and_operation_if_current(GtkTreeModel *tree_model, GtkTreePath *path, GtkTreeIter *iter, gpointer no)
+{
+  const gchar *current_name = ((name_and_operation*)no)->name;
+  const gchar *current_operation = ((name_and_operation*)no)->operation;
+
+  return _delete_if_current(tree_model, path, iter, current_name, current_operation);
+}
+
+static gboolean delete_if_current(GtkTreeModel *tree_model, GtkTreePath *path, GtkTreeIter *iter, gpointer g)
+{
+  const gchar *current_name = gtk_entry_get_text(((dt_gui_presets_edit_dialog_t*)g)->name);
+  const gchar *current_operation = ((dt_gui_presets_edit_dialog_t*)g)->operation;
+
+  return _delete_if_current(tree_model, path, iter, current_name, current_operation);
+}
+
+static gboolean actualize_if_current(GtkTreeModel *tree_model, GtkTreePath *path, GtkTreeIter *iter, gpointer g_pixbuf)
+{
+  GdkPixbuf *check_pixbuf = ((g_and_pixbuf*)g_pixbuf)->check_pixbuf;
+  GdkPixbuf *lock_pixbuf = ((g_and_pixbuf*)g_pixbuf)->lock_pixbuf;
+  dt_gui_presets_edit_dialog_t *g = ((g_and_pixbuf*)g_pixbuf)->g;
+
+  gchar *ith_name = NULL, *ith_operation = NULL;
+  gtk_tree_model_get(tree_model, iter, P_NAME_COLUMN, &ith_name, P_OPERATION_COLUMN, &ith_operation, -1);
+  const gchar *current_name = gtk_entry_get_text(((dt_gui_presets_edit_dialog_t*)g)->name);
+  const gchar *current_operation = ((dt_gui_presets_edit_dialog_t*)g)->operation;
+  if(*ith_name == *current_name && *ith_operation == *current_operation)
+  {
+    g_free(ith_name);
+    g_free(ith_operation);
+    sqlite3_stmt *stmt;
+
+    DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                                "SELECT rowid, name, operation, autoapply, model, maker, lens, iso_min, "
+                                "iso_max, exposure_min, exposure_max, aperture_min, aperture_max, "
+                                "focal_length_min, focal_length_max, writeprotect FROM data.presets WHERE "
+                                "operation = ?1 AND name = ?2",
+                                -1, &stmt, NULL);
+    DT_DEBUG_SQLITE3_BIND_TEXT(stmt, 1, (char*)current_operation, -1, SQLITE_TRANSIENT);
+    DT_DEBUG_SQLITE3_BIND_TEXT(stmt, 2, (char*)current_name, -1, SQLITE_TRANSIENT);
+    while(sqlite3_step(stmt) == SQLITE_ROW)
+    {
+      _update_tree_model_line(tree_model, iter, stmt, lock_pixbuf, check_pixbuf, NULL, *iter, FALSE);
+
+      const gint rowid = sqlite3_column_int(stmt, 0);
```

Comment:
BTW, this code seems very similar to the other above, probably good to do some refactoring and introduce a new common routine.